src/services/paras/ParasService.ts

```diff
@@ -0,0 +1,441 @@
+	 */
+	async leasesCurrent(
+		hash: BlockHash,
+		includeCurrentLeaseHolders: boolean
+	): Promise<ILeasesCurrent> {
+		let blockNumber, currentLeaseHolders;
+		if (!includeCurrentLeaseHolders) {
+			const { number } = await this.api.rpc.chain.getHeader(hash);
+			blockNumber = number.unwrap();
+		} else {
+			const [{ number }, leaseEntries] = await Promise.all([
+				this.api.rpc.chain.getHeader(hash),
+				this.api.query.slots.leases.entriesAt<
+					Vec<Option<ITuple<[AccountId, BalanceOf]>>>,
+					[ParaId]
+				>(hash),
+			]);
+
+			blockNumber = number.unwrap();
+
+			currentLeaseHolders = leaseEntries
+				.filter(([_k, leases]) => leases[0].isSome)
+				.map(([key, _l]) => key.args[0]);
+		}
+
+		const leasePeriod = this.api.consts.slots.leasePeriod as BlockNumber;
+		const leasePeriodIndex = this.currentLeasePeriodIndex(blockNumber);
+		const endOfLeasePeriod = leasePeriodIndex.mul(leasePeriod).add(leasePeriod);
+
+		return {
+			at: {
+				hash,
+				height: blockNumber.toString(10),
+			},
+			leasePeriodIndex,
+			endOfLeasePeriod,
+			currentLeaseHolders,
+		};
+	}
+
+	/**
+	 * List all registered paras (parathreads & parachains).
+	 *
+	 * @param hash `BlockHash` to make call at
+	 * @returns all the current registered paraIds and their lifecycle status
+	 */
+	async paras(hash: BlockHash): Promise<IParas> {
+		const [{ number }, paraLifecycles] = await Promise.all([
+			this.api.rpc.chain.getHeader(hash),
+			this.api.query.paras.paraLifecycles.entriesAt<ParaLifecycle, [ParaId]>(
+				hash
+			),
+		]);
+
+		const parasPromises = paraLifecycles.map(async ([k, paraLifecycle]) => {
+			const paraId = k.args[0];
+			let onboardingAs: ParaType | undefined;
+			if (paraLifecycle.isOnboarding) {
+				const paraGenesisArgs = await this.api.query.paras.paraGenesisArgs.at<ParaGenesisArgs>(
+					hash,
+					paraId
+				);
+				onboardingAs = paraGenesisArgs.parachain.isTrue
+					? 'parachain'
+					: 'parathread';
+			}
+
+			return {
+				paraId,
+				paraLifecycle,
+				onboardingAs,
+			};
+		});
+
+		return {
+			at: {
+				hash,
+				height: number.unwrap().toString(10),
+			},
+			paras: await Promise.all(parasPromises),
+		};
+	}
+
+	/**
+	 * Calculate the current lease period index.
+	 *
+	 * @param blockHeight current blockheight
+	 * @param leasePeriod duration of lease period
+	 */
+	private currentLeasePeriodIndex(now: BN): BN {
+		const leasePeriod = this.api.consts.slots.leasePeriod as BlockNumber;
+		return now.div(leasePeriod);
+	}
+
+	/**
+	 * The offset into the ending samples of the auction. When we are not in the
+	 * ending phase of the auction we can use 0 as the offset, but we do not return
+	 * that here in order to closely mimic `Auctioneer::is_ending` impl in
+	 * polkadot's `runtime::common::auctions`.
+	 *
+	 * @param now current block number
+	 * @param startEnd block number of the start of the auctions ending period
+	 */
+	private endingOffset(now: BN, begginEnd: IOption<BN>): IOption<BN> {
```

Comment:
```suggestion
	 * @param startEnd block number of the start of the auction's ending period
	 */
	private endingOffset(now: BN, startEnd: IOption<BN>): IOption<BN> {
```

Reply:
While I think `startEnd` is better here and matches the comment I am changing to `beginEnd` because that is what we use in the `auctions/current` response. (I originally chose begin instead of start because for auction phase I use `starting` & `ending`, so I thought it would be less confusing if I used different terms).